include/z64game.h

```diff
@@ -0,0 +1,103 @@
+#ifndef Z64_GAME_H
+#define Z64_GAME_H
+
+#include "ultra64.h"
+#include "libc/stdbool.h"
+#include "libc/stdint.h"
+#include "io/controller.h"
+#include "tha.h"
+
+struct GraphicsContext;
+struct GameState;
+
+
+typedef void (*GameStateFunc)(struct GameState* gameState);
+
+typedef struct {
+    /* 0x00 */ void*         loadedRamAddr;
+    /* 0x04 */ uintptr_t     vromStart; // if applicable
+    /* 0x08 */ uintptr_t     vromEnd;   // if applicable
+    /* 0x0C */ void*         vramStart; // if applicable
+    /* 0x10 */ void*         vramEnd;   // if applicable
+    /* 0x14 */ UNK_PTR       unk_14;
+    /* 0x18 */ GameStateFunc init;    // initializes and executes the given context
+    /* 0x1C */ GameStateFunc destroy; // deconstructs the context, and sets the next context to load
+    /* 0x20 */ UNK_PTR       unk_20;
+    /* 0x24 */ UNK_PTR       unk_24;
+    /* 0x28 */ UNK_TYPE      unk_28;
+    /* 0x2C */ size_t        instanceSize;
+} GameStateOverlay; // size = 0x30
+
+typedef struct GameAllocEntry {
+    /* 0x0 */ struct GameAllocEntry* next;
+    /* 0x4 */ struct GameAllocEntry* prev;
+    /* 0x8 */ size_t size;
+    /* 0xC */ u32 unk_0C;
+} GameAllocEntry; // size = 0x10
+
+typedef struct GameAlloc {
+    /* 0x00 */ GameAllocEntry base;
+    /* 0x10 */ GameAllocEntry* head;
+} GameAlloc; // size = 0x14
+
+typedef struct GameState {
+    /* 0x00 */ struct GraphicsContext* gfxCtx;
+    /* 0x04 */ GameStateFunc main;
+    /* 0x08 */ GameStateFunc destroy;
+    /* 0x0C */ GameStateFunc init; // Usually the current game state's init, though after stopping, the graph thread will look here to determine the next game state to load.
+    /* 0x10 */ size_t size;
+    /* 0x14 */ Input input[4];
```

Comment:
```suggestion
    /* 0x14 */ Input input[MAXCONTROLLERS];
```
Already done in https://github.com/zeldaret/mm/pull/1246, but just so it doesn't get lost